pycalphad/core/patched_piecewise.py

```diff
@@ -0,0 +1,149 @@
+import sympy.functions.elementary.piecewise
+from sympy.core import S, Function, Dummy
+from sympy.core.relational import Relational, _canonical
+from sympy.logic.boolalg import And, Boolean, distribute_and_over_or, Or
+from sympy.utilities.misc import filldedent
+
+
+def piecewise_eval(cls, *_args):
+    if not _args:
+        return
+
+    if len(_args) == 1 and _args[0][-1] == True:
+        return _args[0][0]
+
+    newargs = []  # the unevaluated conditions
+    current_cond = set()  # the conditions up to a given e, c pair
+    # make conditions canonical
+    args = []
+    for e, c in _args:
+        if not c.is_Atom and not isinstance(c, Relational):
+            free = c.expr_free_symbols
```

Comment:
This is the only line that was changed compared to the sympy 1.2 base; originally it was `c.free_symbols`. Details in https://github.com/sympy/sympy/issues/14933